front/src/utils/hooks/useDeploymentSettings.ts

```diff
@@ -0,0 +1,90 @@
+import { useEffect, useState } from 'react';
+
+import defaultLogo from 'assets/logo-color.svg';
+import defaultOsrdLogo from 'assets/logo-osrd-color-white.svg';
+import proudLogo from 'assets/proud-logo-color.svg';
+import proudOsrdLogo from 'assets/proud-logo-osrd-color-white.svg';
+import xmasLogo from 'assets/xmas-logo-color.svg';
+import xmasOsrdLogo from 'assets/xmas-logo-osrd-color-white.svg';
+
+const MONTH_VALUES = {
+  JUNE: 5,
+  DECEMBER: 11,
+};
+type DeploymentSettings = {
+  digitalTwinName: string;
+  digitalTwinLogo: string;
+  digitalTwinLogoWithName: string;
+  stdcmName: string;
+  stdcmLogo?: string;
+  stdcmSimulationSheetLogo?: string;
+  isCustomizedDeployment: boolean;
+};
+
+const useDeploymentSettings = () => {
+  const [customizedDeploymentSetting, setCustomizedDeploymentSetting] =
+    useState<DeploymentSettings>({
+      digitalTwinName: 'Osrd',
+      digitalTwinLogo: defaultLogo,
+      digitalTwinLogoWithName: defaultOsrdLogo,
+      stdcmName: 'Stdcm',
+      stdcmLogo: undefined,
+      stdcmSimulationSheetLogo: undefined,
+      isCustomizedDeployment: false,
+    });
+
+  useEffect(() => {
+    const fetchInternalProd = async () => {
+      try {
+        const response = await fetch('/overrides/overrides.json');
+        if (!response.ok || response.headers.get('Content-Type') !== 'application/json') {
+          setCustomizedDeploymentSetting((prev) => ({
+            ...prev,
+            isCustomizedDeployment: false,
+          }));
+
+          if (new Date().getMonth() === MONTH_VALUES.JUNE) {
+            setCustomizedDeploymentSetting((prev) => ({
+              ...prev,
+              digitalTwinLogo: proudLogo,
+              digitalTwinLogoWithName: proudOsrdLogo,
+            }));
+          }
+          if (new Date().getMonth() === MONTH_VALUES.DECEMBER) {
+            setCustomizedDeploymentSetting((prev) => ({
+              ...prev,
+              digitalTwinLogo: xmasLogo,
+              digitalTwinLogoWithName: xmasOsrdLogo,
+            }));
+          }
```

Comment:
It would be better to have only 1 set at the end of this block of code to avoid extra re-renders
```suggestion
          let digitalTwinLogo = defaultLogo;
          let digitalTwinLogoWithName = defaultOsrdLogo;
          const currentMonth = new Date().getMonth();
          
          if (currentMonth === MONTH_VALUES.JUNE) {
            digitalTwinLogo = proudLogo;
            digitalTwinLogoWithName = proudOsrdLogo;
          }
          else if (currentMonth === MONTH_VALUES.DECEMBER) {
            digitalTwinLogo = xmasLogo;
            digitalTwinLogoWithName = xmasOsrdLogo;
          }
          
          setCustomizedDeploymentSetting((prev) => ({
            ...prev,
            isCustomizedDeployment: false,
            digitalTwinLogo,
            digitalTwinLogoWithName
          }));
```

Reply:
Done